org.eclipse.xtext.xbase/deprecated/org/eclipse/xtext/xbase/formatting/HiddenLeafAccess.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2012, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xbase.formatting;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.xtext.formatting2.regionaccess.ITextRegionAccess;
+import org.eclipse.xtext.nodemodel.ICompositeNode;
+import org.eclipse.xtext.nodemodel.ILeafNode;
+import org.eclipse.xtext.nodemodel.INode;
+import org.eclipse.xtext.parsetree.reconstr.impl.NodeIterator;
+import org.eclipse.xtext.xbase.lib.Functions.Function1;
+import org.eclipse.xtext.xbase.lib.IntegerRange;
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+import org.eclipse.xtext.xbase.lib.ListExtensions;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+
+/**
+ * @deprecated use {@link ITextRegionAccess}
+ */
+@Deprecated
+public class HiddenLeafAccess {
+	@Inject
+	private NodeModelAccess nodeModelAccess;
+	private int offset;
```

Comment:
Why is this a field? It's only used in one method

Reply:
refactoring error